ros_ign_bridge/src/convert/geometry_msgs.cpp

```diff
@@ -106,6 +106,36 @@ convert_ign_to_ros(
   convert_ign_to_ros(ign_msg.orientation(), ros_msg.orientation);
 }
 
+template<>
+void
+convert_ros_to_ign(
+  const geometry_msgs::msg::PoseWithCovariance & ros_msg,
+  ignition::msgs::PoseWithCovariance & ign_msg)
+{
+  convert_ros_to_ign(ros_msg.pose.position, *ign_msg.mutable_pose()->mutable_position());  
+  convert_ros_to_ign(ros_msg.pose.orientation, *ign_msg.mutable_pose()->mutable_orientation());
+  for (const auto & elem : ros_msg.covariance) {
+    ign_msg.mutable_covariance()->add_data(elem);
+  }
+}
+
+template<>
+void
+convert_ign_to_ros(
+  const ignition::msgs::PoseWithCovariance & ign_msg,
+  geometry_msgs::msg::PoseWithCovariance & ros_msg)
+{
+  convert_ign_to_ros(ign_msg.pose().position(), ros_msg.pose.position);
+  convert_ign_to_ros(ign_msg.pose().orientation(), ros_msg.pose.orientation);
+  int data_size = ign_msg.covariance().data_size();
+  if (data_size == 36) {
```

Comment:
How likely is this going to be false? Seems less than ideal.

Reply:
I agree, this is not likely to happen, but I was still worried if this loop will segfault at runtime if the data array in ``ros_msg.covariance[ ]`` exceeds index. There is no explicit check on ``Float_V`` size.

Reply:
Maybe those are not supported by protobuf : https://www.aapelivuorinen.com/blog/2019/07/12/protobuf-arrays/

Reply:
yeah I don't think protobuf supports fixed size arrays

we are on the verge of releasing the ign-msgs file into ign-msgs8. Does this approach seem reasonable to you?

* https://github.com/ignitionrobotics/ign-msgs/pull/237

Reply:
I think it's fine to keep this PR as-is. I was just curious if it was possible to avoid.